metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -752,6 +752,9 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
                             table_dict[table_data.table_path].timestamp
                             < table_data.timestamp
                         ):
+                            table_dict[
```

Comment:
Will this cause any issues for folks currently using the connector? Does S3 state that the schema file will _always_ be the last updated?

Reply:
No no. If Dataset(equivalent to folders) consists of multiple files (say a set of csv files), currently first file encountered while browsing this folder is read and its schema is considered. It is not predictable which file's schema will be considered. With this change, schema will be picked from the file that was last updated - this is with assumption that the file that was last written to folder has uptodate schema.


Ideal implementation could deduce schema based on a sample of files picked from this folder.